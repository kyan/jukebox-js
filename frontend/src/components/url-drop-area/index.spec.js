import React, { Component } from 'react'
import { shallow } from 'enzyme'
import TestBackend from 'react-dnd-test-backend'
import { DragDropContext } from 'react-dnd'
import UrlDropArea from './index'

describe('UrlDropArea', () => {
  let wrapper
  const target = jest.fn()
  const onDrop = jest.fn()

  function wrapInTestContext(DecoratedComponent) {
    return DragDropContext(TestBackend)(
      class TestContextContainer extends Component {
        render() {
          return <DecoratedComponent {...this.props} />;
        }
      }
    );
  }

  describe('render', () => {
    const DropArea = wrapInTestContext(UrlDropArea);
    wrapper = shallow(
      <DropArea
        connectDropTarget={target}
        accepts={ ['__URL__'] }
        onDrop={onDrop}
      />
    )

    it('renders as expected', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
