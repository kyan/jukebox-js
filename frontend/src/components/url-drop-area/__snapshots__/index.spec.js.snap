// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlDropArea render renders as expected 1`] = `
<TestContextContainer
  accepts={
    Array [
      "__URL__",
    ]
  }
  connectDropTarget={[Function]}
  onDrop={[Function]}
/>
`;
